osf_tests/test_registration_moderation_notifications.py

```diff
@@ -18,17 +26,11 @@
 )
 
 from website import mails, settings
+from website.notifications import emails, tasks
+from website.reviews import listeners
 
-from osf.utils.workflows import RegistrationModerationTriggers, RegistrationModerationStates
-
-from osf.utils.notifications import (
-    notify_submit,
-    notify_accept_reject,
-    notify_moderator_registration_requests_withdrawal,
-    notify_reject_withdraw_request,
-    notify_withdraw_registration
-)
 
+@mock.patch('website.mails.settings.USE_EMAIL', False)
```

Comment:
nit: this is more of a consistency think but typically when we are mocking a setting like this and we want to test the service behind it we actually set the "enabled bool" to True for the test state. So for a example `share_enabled` and `datacite_enabled` are set to False locally, True on Prod and Staging and True for specific tests, it would be nice if email followed this pattern.

Reply:
Not sure what you're saying. USE_EMAIL being true results in a socket error in tests. I know that it's false in local.py, but I wasn't sure what Travis used, so I wanted to be safe and make sure it's set to False for these tests. Happy to remove the mock here and elsewhere if it's unnecessary

FWIW, USE_EMAIL only governs if an actual email is sent: https://github.com/CenterForOpenScience/osf.io/blob/develop/website/mails/mails.py#L127

By ensuring it's false, we eliminate the need to mock `send_mail`, which (along with store_emails) is one of the only places where we actually attempt to render the email body with the context.

Reply:
Yeah, probably not practical under the circumstances I'm trying to apply some consistency, what I would like to do would be to mock the actual mailer in this situation and then flip the `USE_EMAIL` setting to True (thus avoiding the socket error) But outside of specific email tests, it would be a noop locally and in most travis tests.